bodo/libs/streaming/union.py

```diff
@@ -58,6 +55,28 @@ def __init__(
     def key(self):
         return (self.all, self.in_table_types)
 
+    def is_precise(self):
+        return not (
+            (len(self.in_table_types) == 0)
+            or any(t == types.unknown for t in self.in_table_types)
+        )
+
+    def unify(self, typingctx, other):
```

Comment:
It looks like this would be the only subclass that is required to implement its own unify if you move it to the parent class.

Reply:
See above.